### PR TITLE
Implement remote control in web interface by apache guacamole integration

### PR DESCRIPTION
**Pull request by daanfs**
What's in a title.

### CODE REVIEW

**Comment by JamieH** [comment]
I've just given this a go, It doesn't currently support VNC, I had to install `libvncserver-dev`, It also doesn't work when proxying through nginx, also there's no KVM support currently - but apart from that it works well 👍 Thanks for implementing this

File: cuckoo/web/controllers/analysis/control/api.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+from cuckoo.web.utils import csrf_exempt, json_error_response, api_post
+from django.http import HttpResponse, StreamingHttpResponse, JsonResponse
+from guacamole.client import GuacamoleClient, GuacamoleError
+
+mdb = MongoDB()
+mdb.init_once()
+db = Database()
+log = logging.getLogger(__name__)
+
+sockets = {}
```
Comment (JamieH) [inline cuckoo/web/controllers/analysis/control/api.py]: Looks like this is causing an issue when running via uwsgi.

```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/Cuckoo-2.0.5.3-py2.7.egg/cuckoo/web/controllers/analysis/control/api.py", line 185, in content
    guac = sockets[cache_key]
KeyError: 'bf39c124-cf6a-46fd-9b16-7106feef4384'
```

**Comment by jbremer** [comment]
This has now been merged, closing, thanks :-)

**Comment by marcelmalware** [comment]
Hi @daanfs, how would i go about implementing this with ESX machinery and multiple guest vms?

**Pull request closed without merging**